src/proof/correct_key.rs

```diff
@@ -148,19 +148,19 @@ impl ProveCorrectKey<EncryptionKey, DecryptionKey> for Paillier
             .collect();
 
         // check a co-prime with n
-        if a.par_iter().any(|ai| BigInt::egcd(&dk.n, ai).0 != BigInt::one()) {
-            return Err(ProofError)
-        }
+        fail = a.par_iter()
+            .any(|ai| BigInt::egcd(&dk.n, ai).0 != BigInt::one()) || fail;
 
         // check that e was computed correctly
         let e = compute_digest(
             iter::once(&dk.n)
                 .chain(&challenge.x)
                 .chain(&a)
         );
-        if challenge.e != e {
-            return Err(ProofError)
-        }
+
+        fail = (challenge.e != e) || fail;
+
+        if fail { return Err(ProofError); }
```

Comment:
This is a branch in the code. I think we can remove this branch to the final return